src/components/Api.js

```diff
@@ -0,0 +1,162 @@
+export default class Api {
+  constructor(apiSettings) {
+    this._apiSettings = apiSettings;
+    this._headers = apiSettings.headers;
+    this._baseUrl = apiSettings.baseUrl;
+    this._userInfoUrl = `${this._baseUrl}/users/me`;
+    this._cardsUrl = `${this._baseUrl}/cards`;
+    this._avatarUrl = `${this._userInfoUrl}/avatar`;
+  }
+
+  _checkResponse(response) {
+    if (!response.ok) {
+      throw response;
+    }
```

Comment:
According to the task you need to check if the response is `ok` and use `Promise.reject` to throw a special message with status


```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```